receiver/awscontainerinsightreceiver/internal/neuron/neuron_empty_metric_decorator.go

```diff
@@ -0,0 +1,159 @@
+			} else {
+				populateNonCoreMetrics(metrics, k, attributeConfig[k], hardwareInfo)
+			}
+		}
+
+	}
+}
+
+// method populates per non-core metrics, thus empty metrics are added only once per node
+func populateNonCoreMetrics(metrics pmetric.MetricSlice, metricName string, attributesToAdd []string, hardwareInfo pmetric.Metric) {
+	metricToAdd := createNewMetricFromHardwareInfo(hardwareInfo, metricName)
+	metricBody := metricToAdd.Gauge().DataPoints().At(0)
+
+	for _, attribute := range attributesToAdd {
+		metricBody.Attributes().PutStr(attribute, nonCoreAttributeValues[attribute])
+	}
+
+	metricToAdd.CopyTo(metrics.AppendEmpty())
+}
+
+// method populates per core metrics, thus empty metrics are added per core
+func populateCoreMetrics(metrics pmetric.MetricSlice, metricName string, attributesToAdd []string, hardwareInfo pmetric.Metric) {
+	neuronCoresPerDevice, foundCoresPerDevice := getNeuronCoresPerDevice(hardwareInfo)
+	neuronDeviceCount, foundDeviceCount := getNeuronDeviceCount(hardwareInfo)
+	if foundCoresPerDevice && foundDeviceCount {
+		for coreIndex := 0; coreIndex < neuronCoresPerDevice; coreIndex++ {
+			for deviceNum := 0; deviceNum < neuronDeviceCount; deviceNum++ {
+				metricToAdd := createNewMetricFromHardwareInfo(hardwareInfo, metricName)
+				metricBody := metricToAdd.Gauge().DataPoints().At(0)
+
+				for _, attribute := range attributesToAdd {
+					if attribute == neuronCoreAttributeKey {
+						metricBody.Attributes().PutStr(attribute, strconv.Itoa(coreIndex*neuronDeviceCount+deviceNum))
+					}
+					if attribute == neuronDeviceAttributeKey {
+						metricBody.Attributes().PutStr(attribute, strconv.Itoa(deviceNum))
+					}
+				}
+				metricToAdd.CopyTo(metrics.AppendEmpty())
+			}
+		}
+	}
+}
+
+// returns the device count for neuron from the hardwareInfo metric
+// https://awsdocs-neuron.readthedocs-hosted.com/en/latest/tools/neuron-sys-tools/neuron-monitor-user-guide.html#neuron-hw-counters
+func getNeuronDeviceCount(hardwareInfo pmetric.Metric) (int, bool) {
+	neuronCoreHardwareInfoDatapoints := hardwareInfo.Sum().DataPoints()
+	if neuronCoreHardwareInfoDatapoints.Len() > 0 {
+		neuronDeviceCountValue, found := neuronCoreHardwareInfoDatapoints.At(0).Attributes().Get(neuronDeviceCountAttributeKey)
+		if found {
+			neuronDeviceCount, _ := strconv.Atoi(neuronDeviceCountValue.AsString())
+			return neuronDeviceCount, true
+		}
+	}
+	return -1, false
+}
+
+// returns a empty gauge metric with all attributes of hardwareInfo metric copied
+func createNewMetricFromHardwareInfo(hardwareInfo pmetric.Metric, metricName string) pmetric.Metric {
+	metricToAdd := pmetric.NewMetric()
+	metricToAdd.SetEmptyGauge()
+	hardwareInfo.Sum().DataPoints().CopyTo(metricToAdd.Gauge().DataPoints())
+
+	metricToAdd.SetName(metricName)
+	metricBody := metricToAdd.Gauge().DataPoints().At(0)
+	metricBody.SetDoubleValue(0)
+	metricBody.Attributes().PutStr("runtime_tag", "default")
```

Comment:
To confirm, "default" is a known special value for runtime_tag?

Reply:
Yes correct